extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientProducer.java

```diff
@@ -56,41 +57,39 @@
 
     private void initialize() {
         log.debug("Initializing CacheManager");
-        Configuration conf;
         if (properties == null) {
-            // We already loaded and it wasn't present - so use an empty config
-            conf = new ConfigurationBuilder().build();
-        } else {
-            conf = builderFromProperties(properties).build();
+            // We already loaded and it wasn't present - so don't initialize the cache manager
+            return;
         }
-        cacheManager = new RemoteCacheManager(conf);
 
-        // TODO: do we want to automatically register all the proto file definitions?
-        RemoteCache<String, String> protobufMetadataCache = null;
-
-        Set<SerializationContextInitializer> initializers = (Set) properties.remove(PROTOBUF_INITIALIZERS);
-        if (initializers != null) {
-            for (SerializationContextInitializer initializer : initializers) {
-                if (protobufMetadataCache == null) {
-                    protobufMetadataCache = cacheManager.getCache(
-                            ProtobufMetadataManagerConstants.PROTOBUF_METADATA_CACHE_NAME);
-                }
-                protobufMetadataCache.put(initializer.getProtoFileName(), initializer.getProtoFile());
-            }
-        }
+        Configuration conf = builderFromProperties(properties).build();
+        cacheManager = new RemoteCacheManager(conf);
 
-        for (Map.Entry<Object, Object> property : properties.entrySet()) {
-            Object key = property.getKey();
-            if (key instanceof String) {
-                String keyString = (String) key;
-                if (keyString.startsWith(InfinispanClientProducer.PROTOBUF_FILE_PREFIX)) {
-                    String fileName = keyString.substring(InfinispanClientProducer.PROTOBUF_FILE_PREFIX.length());
-                    String fileContents = (String) property.getValue();
+        if (properties.containsKey(USE_SCHEMA_REGISTRATION) && (Boolean) properties.get(USE_SCHEMA_REGISTRATION)) {
```

Comment:
```suggestion
        InfinispanClientRuntimeConfig infinispanClientRuntimeConfig = this.infinispanClientRuntimeConfig.get();
        if (infinispanClientRuntimeConfig.useSchemaRegistration.orElse(Boolean.TRUE)) {
```
There is no reason to put this in a property as isn't part of the hotrod-client.properties.